### DISCUSSION TITLE
Ledger Hardware Wallet Support—Developer Preview

### DISCUSSION THREAD

**Discussion by kyranjamie** [blockstack/stacks-wallet-web#2451] [Announcements]
# Ledger Support for the Hiro Wallet

We're launching a developer preview of [Ledger](https://ledger.com/) support for the _Hiro Wallet_, to gather user feedback and catch bugs ahead of the forthcoming stable release of this feature.

Here, you'll find all the information you need to set up the preview build.

> **Warning**
> **This is a beta software release.** Do not test this software with your main cryptocurrency holdings. Use a separate Ledger device for testing.

## Getting started

You'll need to follow a few steps to begin testing out the preview release.

### Install Stacks app for Ledger

**Update**: _01-07-2022_ You can download the Stacks App directly from Ledger Live

<details>
<summary>Old development build instructions</summary>

You’ll need the development build of the _Stacks Ledger app_, developed by our partners at [Zondax](https://zondax.ch/).

> **Note**
> Only the _Ledger Nano S_ and _Ledger Nano S Plus_ are supported for preview builds

**Installation instructions**

1. [Download the latest release on their Github, `v0.22.7`](https://github.com/Zondax/ledger-blockstack/releases). There are separate builds depending on which device you have.
1. In your terminal, give the downloaded binary executable permissions `chmod +x installer_nanos.sh`
1. Ensure your test Ledger device is plugged in, unlocked, and on the home screen
1. Run `./installer_nanos.sh load`
1. Follow the instructions that appear on your Ledger device
1. Open the Stacks app (you'll need to approve an _unofficial app_ warning when opening)

</details>

### Install Hiro Wallet developer preview

[Download the latest developer preview build of the Hiro Wallet here →](https://github.com/hirosystems/stacks-wallet-web/actions/runs/2415090218)

Once downloaded, unpack the `.zip` file, and load the *unpacked* extension into your browser.

> **Note**
> For instructions on how to load an “unpacked” Web Extension, [follow Chrome’s developer documentation](https://developer.chrome.com/docs/extensions/mv3/getstarted/#unpacked)

For testing purposes, we recommend you use a different browser to the one you have your primary wallet set up in.

## Upgrading your apps

There are some limitations and differences to be aware of using the _Hiro Wallet_ when set up with a Ledger. 

### `@stacks/connect`

If using [`@stacks/connect`](https://github.com/hirosystems/connect/) you'll need to upgrade to the latest version to allow Ledger users to initiate authentication and transaction signing.

```bash
yarn upgrade @stacks/connect @stacks/connect-react -L

npm update @stacks/connect @stacks/connect-react

```

### No `appPrivateKey`, Gaia support

In software wallet mode, the `appPrivateKey` is derived from the wallet's `dataPrivateKey`. By design, this **is not accessible** when using a hardware wallet.

Apps that rely on the `appPrivateKey`, and features that need this key, such as Gaia, will not work. We're aware this has a big impact for some projects. If this affects your app, let us know, and [contribute to this discussion to consider alternative implementations →](https://github.com/hirosystems/stacks-wallet-web/discussions/2352)

### No Message Signing support

This is a new feature released in [`v3.8`](https://github.com/hirosystems/stacks-wallet-web/releases/tag/v3.8.0) and is not currently supported in Ledger mode.

## Feedback

💬 If you have high-level thoughts, feedback, or questions on our Ledger implementation, comment here on the discussion 👇🏼 

🐞 For bugs, [create a new issue here →](https://github.com/hirosystems/stacks-wallet-web/issues/new?labels=ledger&title=Bug%20found%20testing%20ledger%20where%20%3Cdelete%20this%20and%20describe%20issue%3E)

We look forward to everyone's input,
Hiro UserX team ❤️

**Comment by fbwoolf** [comment]
The latest build is now `v0.22.7`: https://github.com/Zondax/ledger-blockstack/releases
